### PR TITLE
[prometheus-nginx-exporter] Fix Annotation Use in Chart

### PR DESCRIPTION
**Pull request by lukasklinger**
#### What this PR does / why we need it
The prometheus-nginx-exporter has two ways to add additional annotations to the resources created by it, the `additionalAnnotations` field and the `podAnnotations` field. Using them in the current version of the chart either breaks the templating or - in the case of the `podAnnotations` field - has no effect.

This PR unifies the use of the `additionalAnnotations` field across the different resources and integrates the `podAnnotations` field into a new helper so that annotations added in this field have an effect.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-nginx-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -4,10 +4,7 @@ kind: Deployment
 metadata:
   name: {{ include "prometheus-nginx-exporter.fullname" . }}
   namespace: {{ template "prometheus-nginx-exporter.namespace" . }}
-  {{- with .Values.additionalAnnotations }}
-  annotations:
-  {{ toYaml . | indent 4 }}
-  {{- end }}
+  annotations: {{ toYaml .Values.additionalAnnotations | nindent 4 }}
```
Comment (zeritti) [inline charts/prometheus-nginx-exporter/templates/deployment.yaml:7]: There is nothing wrong with the previous code. In fact, we should prefer not to render an empty field. Applies to all the changes.

```yaml
{{- with .Values.additionalAnnotations }}
annotations: {{ toYaml . | nindent 4 }}
{{- end }}
```

File: charts/prometheus-nginx-exporter/templates/role.yaml [hunk 2]
```diff
@@ -5,10 +5,7 @@ apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: {{ template "prometheus-nginx-exporter.fullname" . }}
   namespace: {{ template "prometheus-nginx-exporter.namespace" . }}
-  {{- if .Values.additionalAnnotations }}
-  annotations:
-  {{ toYaml .Values.additionalAnnotations | indent 4 }}
-  {{- end }}
+  annotations: {{ toYaml .Values.additionalAnnotations | nindent 4 }}
```
Comment (zeritti) [inline charts/prometheus-nginx-exporter/templates/role.yaml:8]:
```yaml
{{- with .Values.additionalAnnotations }}
annotations: {{ toYaml . | nindent 4 }}
{{- end }}
```

File: charts/prometheus-nginx-exporter/templates/serviceaccount.yaml [hunk 3]
```diff
@@ -3,10 +3,7 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  {{- if .Values.additionalAnnotations }}
-  annotations:
-  {{ toYaml .Values.additionalAnnotations | indent 4 }}
-  {{- end }}
+  annotations: {{ toYaml .Values.additionalAnnotations | nindent 4 }}
```
Comment (zeritti) [inline charts/prometheus-nginx-exporter/templates/serviceaccount.yaml:6]:
```yaml
{{- with .Values.additionalAnnotations }}
annotations: {{ toYaml . | nindent 4 }}
{{- end }}
```

File: charts/prometheus-nginx-exporter/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -5,10 +5,7 @@ kind: ServiceMonitor
 metadata:
   name: {{ template "prometheus-nginx-exporter.fullname" .  }}
   namespace: {{ template "prometheus-nginx-exporter.namespace" . }}
-  {{- if .Values.additionalAnnotations }}
-  annotations:
-  {{ toYaml .Values.additionalAnnotations | indent 4 }}
-  {{- end }}
+  annotations: {{ toYaml .Values.additionalAnnotations | nindent 4 }}
```
Comment (zeritti) [inline charts/prometheus-nginx-exporter/templates/servicemonitor.yaml:8]:
```yaml
{{- with .Values.additionalAnnotations }}
annotations: {{ toYaml . | nindent 4 }}
{{- end }}
```

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @lukasklinger, for your PR. Please, see my comments below.

**Comment by lukasklinger** [comment]
Thanks for the review! I've integrated the feedback 🙂

**Review by zeritti — APPROVED** [review]

**Pull request merged**